### PR TITLE
HIP build corrections

### PR DESCRIPTION
**Pull request by r-abishekmcw**
- Fix HIP backend issues
- Changes for hip build
- Change cmakelists for linking issues
- Change include hip/hip_hcc.h to hip/hip_ext.h to avoid warning

### CODE REVIEW

**Comment by kiritigowda** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Clones removed
==============
+ src/modules/hip/hip_morphological_transforms.cpp  -17
+ src/modules/hip/hip_color_model_conversions.cpp  -27
+ src/modules/hip/hip_imgaugmentation.cpp  -19
+ src/modules/hip/hip_filter_functions.cpp  -6
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/rpp/pullRequest?prid=6412277&bid=20779672)

**Comment by kiritigowda** [comment]
@LakshmiKumar23 can you check this PR

**Review by LakshmiKumar23 — APPROVED** [review]
@kiritigowda builds successfully on both rocm3.7 and rocm3.8

File: src/modules/hip/hip_color_model_conversions.cpp [hunk 1]
```diff
@@ -86,47 +86,47 @@ color_temperature_hip_batch (   Rpp8u* srcPtr, Rpp8u* dstPtr, rpp::Handle& handl
...
-        std::vector<size_t> vgd{(srcSize.width + 31) & ~31, (srcSize.height + 31) & ~31,channel};
-        handle.AddKernel("", "", "vignette.cpp", "vignette_pkd", vld, vgd, "")(srcPtr,
-                                                                            dstPtr,
-                                                                            srcSize.height,
-                                                                            srcSize.width,
-                                                                            channel,
-                                                                            stdDev
-                                                                            );
-    }
+    // int ctr=0;
```
Comment (rrawther) [inline src/modules/hip/hip_color_model_conversions.cpp:89]: if we are not using these implementations anymore, it is better to remove the entire function

Comment (rrawther) [reply src/modules/hip/hip_color_model_conversions.cpp:89]: For all the commented functions in this file

File: src/modules/hip/hip_imgaugmentation.cpp [hunk 2]
```diff
@@ -377,57 +377,57 @@ noise_hip(Rpp8u* srcPtr, RppiSize srcSize, Rpp8u* dstPtr,
...
-        // clSetKernelArg(theKernel, ctr++, sizeof(unsigned int), &channel);
-        // clSetKernelArg(theKernel, ctr++, sizeof(unsigned int), &pixelDistance);
-
-        // size_t gDim3[3];
-        // gDim3[0] = srcSize.width;
-        // gDim3[1] = srcSize.height;
-        // gDim3[2] = 1;
-        // cl_kernel_implementer (handle, gDim3, NULL/*Local*/, theProgram, theKernel);
-    }
+    // srand(time(0));
```
Comment (rrawther) [inline src/modules/hip/hip_imgaugmentation.cpp:380]: Is this not used anymore? Better to remove the function all together

File: src/modules/hip/hip_morphological_transforms.cpp [hunk 3]
```diff
@@ -90,47 +90,47 @@ dilate_hip_batch (   Rpp8u* srcPtr, Rpp8u* dstPtr, rpp::Handle& handle,
...
-        std::vector<size_t> vgd{(srcSize.width + 31) & ~31, (srcSize.height + 31) & ~31,channel};
-        handle.AddKernel("", "", "erode.cpp", "erode_pln", vld, vgd, "")(srcPtr,
-                                                                dstPtr,
-                                                                srcSize.height,
-                                                                srcSize.width,
-                                                                channel,
-                                                                kernelSize
-                                                                );
-    }
+    // int ctr=0;
```
Comment (rrawther) [inline src/modules/hip/hip_morphological_transforms.cpp:93]: please remove all the irrelevant code

**Review by rrawther — COMMENTED** [review]
A lot of functions with commented code and returning SUCCESS. It is better to remove those if not needed anymore.

**Pull request merged**